### PR TITLE
[WIP][br] add APIs for adding OMR via platform generated RA

### PR DESCRIPTION
**Pull request by erjiaqing**
This PR adds the following new APIs

- otBorderRoutingAddPrefixByRouterAdvertisement
- otBorderRoutingSetAcceptingRouterAdvertisementEnabled
- otBorderRoutingIsAcceptingRouterAdvertisementEnabled

They are used to support the DHCPv6 PD on Thread Border Routers. On a typical OS, the DHCPv6 Client of the platform network stack will request a IPv6 subnet via DHCPv6-PD, then invoke the daemon for sending router advertisements on the given interface.

With this PR, the RA will be catched by the `processTransmit` in netif.cpp, which will then pass the message to `otBorderRoutingAddPrefixByRouterAdvertisement`, `RoutingManager` will then add the prefix according to the PIO in the router advertisement, and take care of the lifetime of the prefix added.

### CODE REVIEW

File: include/openthread/border_routing.h [hunk 1]
```diff
@@ -326,6 +326,59 @@ otError otBorderRoutingGetNextPrefixTableEntry(otInstance
...
+ *
+ * @param[in] aInstance A pointer to an OpenThread instance.
+ * @param[in] aMessage  A pointer to an ICMPv6 RouterAdvertisement message.
+ * @param[in] aLength   The length of ICMPv6 RouterAdvertisement message.
+ *
+ * @retval OT_ERROR_NONE          Successfully processed the prefix infomation in the message.
+ * @retval OT_ERROR_INVALID_STATE Routing manager is configured to not handling RA.
+ *
+ */
+otError otBorderRoutingAddPrefixByRouterAdvertisement(otInstance *aInstance, const uint8_t *aMessage, uint16_t aLength);
```
Comment (abtink) [inline include/openthread/border_routing.h:347]: Can we instead use the `otPlatInfraIfRecvIcmp6Nd()` to pass the RA message back to OT?
- We can use `aSrcAddress` such that it is the device's address.
- Then in `RoutingManager` we can tell if this RA is generated by platform by checking the address using `otPlatInfraIfHasAddress()` (and process it differently)?

This way we don't need to add new platform API.

Comment (wgtdkp) [reply include/openthread/border_routing.h:347]: @abtink RA in `otPlatInfraIfRecvIcmp6Nd()` is message received on the infra interface, but the RA here is received on the Thread tunnel interface. I don't think it's going to work

Comment (abtink) [reply include/openthread/border_routing.h:347]: I see. Make sense. Thanks.

**Comment by erjiaqing** [comment]
Close initial PoC PR for PD

**Pull request closed without merging**